src/proof/arith_proof.cpp

```diff
@@ -73,15 +75,22 @@ void ProofArith::toStream(std::ostream& out) {
   toStreamLFSC(out, ProofManager::getArithProof(), d_proof, map);
 }
 
-void ProofArith::toStreamLFSC(std::ostream& out, TheoryProof * tp, theory::eq::EqProof * pf, const ProofLetMap& map) {
+void ProofArith::toStreamLFSC(std::ostream& out, TheoryProof * tp,
+                              std::shared_ptr<theory::eq::EqProof> pf,
```

Comment:
This (and other `toStreamLFSC` methods) should probably use a `const std::weak_ptr` since `toStreamLFSC` is not taking (shared) ownership.